### PR TITLE
feat: endpoints to new users panel

### PR DESCRIPTION
**Pull request by tapiarafael**
## Further comments

[WM-185](https://rocketchat.atlassian.net/browse/WM-185)

[WM-185]: https://rocketchat.atlassian.net/browse/WM-185?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/01-users.js [hunk 1]
```diff
@@ -4229,4 +4229,108 @@ describe('[Users]', function () {
...
+				.get(api('users.list/abcd'))
+				.set(credentials)
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body.errorType).to.be.equal('error-invalid-status');
+					expect(res.body.error).to.be.equal('Invalid status parameter [error-invalid-status]');
+				});
+		});
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/01-users.js:4334]: Can we add some cases for the permissions as well?

File: packages/rest-typings/src/v1/users/UsersListStatusParamsGET.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	properties: {
+		role: {
+			type: 'string',
+		},
+	},
+	required: ['role'],
+	additionalProperties: false,
+};
+
+export const isUsersListStatusProps = ajv.compile<UsersListStatusParamsGET>(UsersListStatusParamsGetSchema);
```
Comment (MarcosSpessatto) [inline packages/rest-typings/src/v1/users/UsersListStatusParamsGET.ts:22]: I think we are forgetting to use this on the endpoint 😬

File: apps/meteor/app/api/server/v1/users.ts [hunk 3]
```diff
@@ -555,6 +556,121 @@ API.v1.addRoute(
...
+
+API.v1.addRoute(
+	'users.sendWelcomeEmail',
+	{
+		authRequired: true,
+	},
+	{
+		async post() {
+			const { email } = this.bodyParams;
+			if (!email) {
```
Comment (MarcosSpessatto) [inline apps/meteor/app/api/server/v1/users.ts:662]: Couldn't we create an Ajv validation for this instead?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/app/api/server/v1/users.ts [hunk 4]
```diff
@@ -555,6 +556,121 @@ API.v1.addRoute(
...
+				return API.v1.failure("The 'email' param is required");
+			}
+
+			if (await sendWelcomeEmail(email)) {
```
[the remark this answers is not in the document]
Comment (tapiarafael) [reply apps/meteor/app/api/server/v1/users.ts:666]: I agree. I used an old file as an example to create mine and didn´t think about it.

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**